feat: merge connector and operator repos into this repo (#72)
instill-ai/component: .github/CONTRIBUTING.md

**Connector**

> - **Connectors** are used for connecting the pipeline to a vendor service. They are defined and initialized in the [connector](https://github.com/instill-ai/component/blob/main/pkg/connector).

Reviewer: ```suggestion
- **Connectors** are used for connecting the pipeline to a vendor service. They are defined and initialized in the [connector](../pkg/connector) package.
```

**Operator**

> - An operator is used for data operations inside the pipeline. They are defined and initialized in the [operator](https://github.com/instill-ai/component/blob/main/pkg/operator).

Reviewer: ```suggestion
- An operator is used for data operations inside the pipeline. They are defined and initialized in the [operator](../pkg/operator) package.
```

2 configuration files define the behaviour of the component:

- `definitions.json`

>     - You can refer to [OpenAI connector](https://github.com/instill-ai/component/blob/main/pkg/connector/openai/v0/config/definitions.json) as an example.

Reviewer: ```suggestion
    - You can refer to [OpenAI connector](../pkg/connector/openai/v0/config/definitions.json) as an example.
```

        This will showcase the upcoming component at [instill.tech](https://instill.tech).
    - We define the `resource_configuration` in this file, which defines the connector resource setup.
- `tasks.json`

>     - You can refer to [OpenAI connector](https://github.com/instill-ai/component/blob/main/pkg/connector/openai/v0/config/tasks.json) as an example.

Reviewer: ```suggestion
    - You can refer to [OpenAI connector](../pkg/connector/openai/v0/config/tasks.json) as an example.
```

<!--
TODO:

>  1. explain how we import the connectors or operators like [here](https://github.com/instill-ai/component/blob/main/pkg/connector/main.go)

Reviewer: ```suggestion
 1. explain how we import the connectors or operators like [here](../pkg/connector/main.go)
```
